test/integration/host_test.rb

```diff
@@ -533,6 +475,29 @@ def test_show_action_buttons
       host.reload
       assert_equal env1.name, host.environment.name
     end
+
+    test 'enables override of hostgroup defaults' do
+      original_environment = FactoryGirl.create(:environment)
+      original_hostgroup = FactoryGirl.create(:hostgroup,
+                                              :environment => original_environment)
+
+      # Clear host environment, then inherit it from hostgroup
+      @host.environment = nil
+      @host.hostgroup = original_hostgroup
+      @host.send(:assign_hostgroup_attributes, ['environment_id'])
```

Comment:
If you want to reassign environment from hostgroup, you need to do it differently:
``` Ruby
@host.attributes = @host.apply_inherited_attributes('hostgroup_id' => original_hostgroup.id)
```